semgrep/semgrep/semgrep_types.py

```diff
@@ -14,97 +17,76 @@
-    BASH: str = "bash"
-    REGEX: str = "regex"
-    GENERIC: str = "generic"
-
-
-class Language_util:
-    language_to_strs: Dict[Language, List[str]] = {
-        Language.PYTHON: [Language.PYTHON.value, "py"],
-        Language.PYTHON2: [Language.PYTHON2.value],
-        Language.PYTHON3: [Language.PYTHON3.value],
-        Language.JAVASCRIPT: [Language.JAVASCRIPT.value, "js"],
-        Language.TYPESCRIPT: [Language.TYPESCRIPT.value, "ts"],
-        Language.JAVA: [Language.JAVA.value],
-        Language.C: [Language.C.value],
-        Language.CPP: [Language.CPP.value, "C++"],
-        Language.GO: [Language.GO.value, "golang"],
-        Language.RUBY: [Language.RUBY.value, "rb"],
-        Language.PHP: [Language.PHP.value],
-        Language.HACK: [Language.HACK.value, "hacklang"],
-        Language.LUA: [Language.LUA.value],
-        Language.CSHARP: [Language.CSHARP.value, "cs", "C#"],
-        Language.RUST: [Language.RUST.value, "Rust", "rs"],
-        Language.KOTLIN: [Language.KOTLIN.value, "Kotlin", "kotlin"],
-        Language.YAML: [Language.YAML.value, "Yaml"],
-        Language.SCALA: [Language.SCALA.value],
-        Language.VUE: [Language.VUE.value],
-        Language.HTML: [Language.HTML.value],
-        Language.ML: [Language.ML.value, "ocaml"],
-        Language.JSON: [Language.JSON.value, "JSON", "Json"],
-        Language.HCL: [Language.HCL.value, "tf", "terraform", "HCL"],
-        Language.BASH: [Language.BASH.value, "sh"],
-        Language.REGEX: [Language.REGEX.value, "none"],
-        Language.GENERIC: [Language.GENERIC.value],
-    }
-
-    str_to_language = {}
-    for language in language_to_strs.keys():
-        for lang_str in language_to_strs[language]:
-            str_to_language[lang_str] = language
-
-    """ Convert an inputted string representing a language to a Language
-
-    Keyword arguments;
-    lang_str -- string representing a language (e.g. "C#")
-    span     -- span of language string in the original file (for error reporting),
-                None if resolve was called within semgrep
+
+class Language(str):
+    pass
+
+
+@dataclass(frozen=True)
+class LanguageDefinition:
     """
+    Mirrors schema of lang.json (see lang/README.md) for each language
+    """
+
+    id: Language
+    name: str
+    keys: Collection[str]
+    exts: Collection[FileExtension]
+    reverse_exts: Collection[str]
+    shebangs: Collection[Shebang]
 
     @classmethod
-    def resolve(cls, lang_str: str, span: Optional[Span] = None) -> Language:
-        if lang_str in cls.str_to_language:
-            return cls.str_to_language[lang_str]
+    def from_dict(cls, data: JsonObject) -> "LanguageDefinition":
+        return cls(
+            id=Language(data["id"]),
+            name=data["name"],
+            keys=data["keys"],
+            exts=data["exts"],
+            reverse_exts=data.get("reverse_exts", data["exts"]),
+            shebangs=data.get("shebangs", []),
+        )
+
+
+class Lang:
```

Comment:
Lang is pretty close to Language and might get confusing. What do you think of `Linguist`?

Also looks like most places LANG (which should also be renamed) is used, it is invoked `LANG.data_by_id[language]` should we just expose a `get_definition_for_language` function so it's more readable?

Reply:
I'll just make the class private, since one only every interacts with the singleton anyway